python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -656,6 +656,45 @@ def is_successful(self) -> bool:
 
         return self.process.wait() == 0
 
+    @public
+    def get_error(self) -> Optional[Exception]:
```

Comment:
Perhaps:

```suggestion
    def raise_on_error(self) -> NoReturn:
```

- Make it raise a `DagsterInvariantException` or something if a user tries to retrieve an error from a successful invocation
- Otherwise `raise` instead of `return`
- Since this function is guaranteed to `NoReturn`, then we should force it consume the entire event stream before return so we give the most verbose error, with error logs parsed.

Reply:
This name would also conflict with the current `raise_on_error: bool` property